spec_parser/owl_to_context.py

```diff
@@ -0,0 +1,112 @@
+    "decisionType",
+    "justificationType",
+    "catalogType",
+    "conditionality",
+    "sbomType",
+    "softwareLinkage",
+    "purpose",
+]
+
+REFERENCE_PROPERTY_TYPES = [
+    "core:Element",
+    "core:Agent",
+]
+
+
+def convert_spdx_owl_to_jsonld_context(spdx_owl: str, out_dir: str):
+    # spdx_owl must point to the OWL in json-ld format
+    with open(spdx_owl, "r") as infile:
+        owl_dict = json.load(infile)
+
+    context_dict = owl_dict["@context"]
+
+    for node in owl_dict["@graph"]:
+        # print(node)
+        node_type = node.get("@type")
+        if not node_type:
+            # print(node)
+            continue
+
+        if "owl:NamedIndividual" in node_type:
+            continue
+        elif node_type in ["owl:DatatypeProperty", "owl:ObjectProperty"]:
+            name = node["@id"].split(":")[-1]
+            type_id = node["rdfs:range"]["@id"]
+
+            if name in context_dict and context_dict[name]["@id"].startswith("core"):
+                # if in doubt, prioritize core properties
+                continue
+
+            if name in PROPERTIES_WITH_ENUM_RANGE:
+                if name == "profile":
+                    # FIXME: since the allowed values for the profile enum collide with
+                    # our namespaces, we need to explicitly remap their meaning in the context
+                    context_dict[name] = {
+                        "@id": node["@id"],
+                        "@type": "@vocab",
+                        "@context": {
+                            "core": "https://spdx.org/rdf/Core/ProfileIdentifierType/core",
+                            "software": "https://spdx.org/rdf/Core/ProfileIdentifierType/software",
+                            "licensing": "https://spdx.org/rdf/Core/ProfileIdentifierType/licensing",
+                            "security": "https://spdx.org/rdf/Core/ProfileIdentifierType/security",
+                            "build": "https://spdx.org/rdf/Core/ProfileIdentifierType/build",
+                            "ai": "https://spdx.org/rdf/Core/ProfileIdentifierType/ai",
+                            "dataset": "https://spdx.org/rdf/Core/ProfileIdentifierType/dataset",
+                            "usage": "https://spdx.org/rdf/Core/ProfileIdentifierType/usage",
+                            "extension": "https://spdx.org/rdf/Core/ProfileIdentifierType/extension",
+                        },
+                    }
+                else:
+                    context_dict[name] = {
+                        "@id": node["@id"],
+                        "@type": "@vocab",
+                        "@context": {"@vocab": type_id + "/"},
+                    }
+            elif node_type == "owl:ObjectProperty" and type_id in REFERENCE_PROPERTY_TYPES:
+                context_dict[name] = {"@id": node["@id"], "@type": "@id"}
+            else:
+                context_dict[name] = {"@id": node["@id"], "@type": type_id}
+
+        elif node_type == "owl:Class":
```

Comment:
maybe use functions, like:

```suggestion
        elif nodeIsClass(node):
```

Reply:
I'm not so sure about that one. A single equivalence check that only occurs once does not seem to justify the extraction.